MainAnalysis/20250611_ChargedHadronRaa_Eventselection/ChargedHadronRAA.cpp

```diff
@@ -0,0 +1,175 @@
+#include <TCanvas.h>
+#include <TCut.h>
+#include <TFile.h>
+#include <TH1D.h>
+#include <TH2D.h>
+#include <TH3D.h>
+#include <TLegend.h>
+#include <TTree.h>
+
+#include <iostream>
+
+using namespace std;
+#include "CommandLine.h"
+#include "Messenger.h"
+#include "ProgressBar.h"
+#include "helpMessage.h"
+#include "parameter.h" // Parameters for the analysis
+#include "utilities.h" // Utility functions for the analysis
+
+//============================================================//
+// Function to check for configuration errors
+//============================================================//
+bool checkError(const Parameters &par) { return false; }
+
+//======= eventSelection =====================================//
+// Check if the event pass eventSelection criteria
+//============================================================//
+bool eventSelection(ChargedHadronRAATreeMessenger *ch, const Parameters &par) {
+
+  // Z Vertex cut 
+  if(fabs(ch->VZ) >= par.VzMax){
+    return false;
+  }
+
+  // NVtx cut
+  if(ch->nVtx < par.NVtxMin) {
+    return false;
+  }
+
+  // Cluster Compatibility Filter
+  if(ch->ClusterCompatibilityFilter < par.CCFilter) {
+    return false;
+  }
+
+  // PV selection
+  if(ch->PVFilter < par.PVFilter) {
+    return false;
+  }
+
+  // Hijing Npart > 1 
```

Comment:
I would not put this here, but in the main code since we might need to revisit this part, and I want to make sure we dont forget about it.